flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannelTest.java

```diff
@@ -448,6 +455,36 @@ public void testUnblockReleasedChannel() throws Exception {
 		localChannel.resumeConsumption();
 	}
 
+	@Test
+	public void testCheckpointingInflightData() throws Exception {
+		SingleInputGate inputGate = new SingleInputGateBuilder().build();
+		List<Buffer> receivedBuffers = new ArrayList<>();
+		inputGate.registerBufferReceivedListener(new BufferReceivedListener() {
+			@Override
+			public void notifyBufferReceived(Buffer buffer, InputChannelInfo channelInfo) {
+				receivedBuffers.add(buffer);
+			}
+
+			@Override
+			public void notifyBarrierReceived(CheckpointBarrier barrier, InputChannelInfo channelInfo) {}
+		});
+
+		ResultSubpartitionView subpartitionView = createResultSubpartitionView(true);
```

Comment:
add maybe one more buffer and some events, like `EndOfPartitionEvent` at the end?